src/struct_mv/struct_vector.c

```diff
@@ -1681,9 +1681,11 @@ hypre_StructVectorRead( MPI_Comm    comm,
    hypre_StructVectorAssemble(vector);
 
    /*----------------------------------------
-    * Close file
+    * Clean up and close file
     *----------------------------------------*/
 
+   HYPRE_StructGridDestroy(grid);
```

Comment:
Why do you destroy the grid here? Don't you want to keep it around in the vector data structure?

Also, answering your comment above in line 1669, it looks like the num ghost is set inside `HYPRE_StructVectorCreate()`, so no need to call `hypre_StructVectorSetNumGhost()` here.

Reply:
I believe a reference to the grid is kept in the vector, so destroying it here really only decrements a reference counter.  I needed to add this to fix a memory leak.

Regarding the num_ghost question, we do pass this into the routine, and there was another issue I had to fix related to passing in an incorrectly sized num_ghost array from the struct driver.  I guess I was wondering if there was a case of reading a vector where we would want to have a differently-defined num_ghost argument (from the default set in Create).

Reply:
Ah, I see the ref counter thing now (I didn't realize it was incremented both on grid creation and vector creation). Thanks!